app/actions/remote/post.ts

```diff
@@ -858,12 +858,13 @@ export const editPost = async (serverUrl: string, postId: string, postMessage: s
 
         const post = await getPostById(database, postId);
         if (post) {
-            const {update_at, edit_at, message: updatedMessage} = await client.patchPost({message: postMessage, id: postId});
+            const {update_at, edit_at, message: updatedMessage, message_source} = await client.patchPost({message: postMessage, id: postId});
             await database.write(async () => {
                 await post.update((p) => {
                     p.updateAt = update_at;
                     p.editAt = edit_at;
                     p.message = updatedMessage;
+                    p.messageSource = message_source || '';
```

Comment:
why empty like this?

perhaps in case is undefined or null, which probably is not what we want?
```suggestion
                    p.messageSource = message_source ?? '';
```

Reply:
The database value is a string, so we want it always to be a string. Null or undefined will have the same meaning as the empty string.
Since the post has been edited, we expect the result from `patchPost` to properly update also the `message_source`. So if it comes undefined, I want to clear whatever we have on the database. If not, changing a message from `Hi! ![](link to image)` to `Bye!` will result in showing `Hi! ![](ink to image)` when editing the post.